src/components/Orderbook.tsx

```diff
@@ -1,196 +1,414 @@
+            })}
+          </Body>
+        </>
+      )}
+    </Table>
+  )
+}
+
+interface OrderbookProps {
+  orders: ObwatchApi.Order[]
+  refresh: (abortCtrl: AbortController) => Promise<void>
+}
+
+export function Orderbook({ orders, refresh }: OrderbookProps) {
+  const { t } = useTranslation()
+  const settings = useSettings()
+  const [search, setSearch] = useState('')
+  const [isLoadingRefresh, setIsLoadingRefresh] = useState(false)
+  const counterpartyCount = useMemo(() => new Set(orders.map((it) => it.counterparty)).size, [orders])
+
+  const tableData: TableTypes.Data = useMemo(() => {
+    const searchVal = search.replace('.', '').toLowerCase()
+    const nodes = orders
+      .filter((order) => {
+        if (search === '') return true
+        return (
+          order.type.toLowerCase().includes(searchVal) ||
+          order.counterparty.toLowerCase().includes(searchVal) ||
+          order.fee.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.minimumSize.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.maximumSize.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.minerFeeContribution.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.bondValue.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.orderId.toLowerCase().includes(searchVal)
+        )
+      })
+      .map((order) => ({
+        ...order,
+        id: `${order.counterparty}_${order.orderId}`,
+      }))
+
+    return { nodes }
+  }, [orders, search])
+
+  return (
+    <div className={styles.orderbookContainer}>
+      <div className={styles.titleBar}>
+        <div className="d-flex justify-content-center align-items-center gap-2">
+          <rb.Button
+            className={styles.refreshButton}
+            variant={settings.theme}
+            onClick={() => {
+              if (isLoadingRefresh) return
+
+              setIsLoadingRefresh(true)
+
+              const abortCtrl = new AbortController()
+              refresh(abortCtrl).finally(() => {
+                // as refreshing is fast most of the time, add a short delay to avoid flickering
+                setTimeout(() => setIsLoadingRefresh(false), 250)
+              })
+            }}
+          >
+            {isLoadingRefresh ? (
+              <rb.Spinner as="span" animation="border" size="sm" role="status" aria-hidden="true" />
+            ) : (
+              <Sprite symbol="refresh" width="24" height="24" />
+            )}
+          </rb.Button>
+          <div className="small">
+            {t('orderbook.text_orderbook_summary', {
+              counterpartyCount,
+              orderCount: orders.length,
+            })}
+            {search && (
+              <>
+                <br />(
+                {t('orderbook.text_orderbook_filter_count', {
+                  filterCount: tableData.nodes.length,
+                })}
+                )
+              </>
+            )}
```

Comment:
https://user-images.githubusercontent.com/10026790/182799404-471a9dcf-9ec2-4b98-9acd-c2cc7b9b3424.mov

Adding that text below looks a bit chunky as the header size changes and the texts don't look aligned since one is in parentheses. What about just replacing the `orderbook.text_orderbook_summary` text with the count of orders that's currently displayed? i.e.:

```
SEARCH: (none)

3 orders by 3 counterparties

ORDER: foobar
ORDER: foobaz
ORDER: bla
```

and

```
SEARCH: foo

2 orders by 2 counterparties

ORDER: foobar
ORDER: foobaz
```

Reply:
Good point!

What do you think of adding a hint if the search is activated?
e.g. "2 orders by 2 counterparties (filtered)",  "found 2 orders by 2 counterparties", "2 orders by 2 counterparties match given search criteria"...

Or do you think it is totally unnecessary?

Reply:
I changed it to have "(filtered)" at the end, as I feel that _some_ sort of visual feedback that the search value has actually been applied is a Good Thing™. However, if you think it is stupid and unncessary, I'll remove it.

Reply:
I like "found 2 orders by 2 counterparties". It feels the most natural to read imo.

Reply:
Pushed the wording update myself, hope that's ok. :)